GDJS/Runtime/bitmapfontmanager.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @constructor
+ * @memberof gdjs
+ * @class BitmapFontManager
+ */
+gdjs.BitmapFontManager = function () {
+  this._fontUsed = {};
+};
+
+gdjs.BitmapFontManager.prototype.setFontUsed = function (name) {
+  if (this._fontUsed[name]) {
+    this._fontUsed[name].countObjectUsingTheFont =
+      this._fontUsed[name].countObjectUsingTheFont + 1 || 1;
+  } else {
+    this._fontUsed[name] = {
+      countObjectUsingTheFont: 1,
+    };
+  }
+};
+
+gdjs.BitmapFontManager.prototype.removeFontUsed = function (name) {
+  if (this._fontUsed[name]) {
+    this._fontUsed[name].countObjectUsingTheFont =
+      this._fontUsed[name].countObjectUsingTheFont - 1 || 0;
+  }
+
+  if (this._fontUsed[name].countObjectUsingTheFont <= 0) {
+    console.log('- Delete the font: ' + name);
+    PIXI.BitmapFont.uninstall(name);
+    delete this._fontUsed[name];
```

Comment:
Seems like you got the idea of the implementation :) 👍
If it works properly, we can then add a final touch, a cache:
- the cache is an array of font names "to be unloaded".
- when `removeFontUsed` is called, if it font reaches 0 usage, instead of immediately deleting it, we put it into the cache of font "to be unloaded".
  - if the cache of font "to be unloaded" is reaching a size > 10, we uninstall the last fonts of the array until we reach again the size 10.
- when `setFontUsed` is called, we check the cache of font "to be unloaded" and if the font name is inside, we remove it (because now there is at least one usage).

Why doing that? It ensures that we have a cache of a maximum of 10 different fonts, that won't get unloaded immediately. If for example you have a unique BitmapText object that uses a font, and that you delete it and re-create it a few seconds later, chance are that the BitmapFont will still be in the cache (unless you deleted tons of other objects). Same if you go from one scene to another: at least 10 fonts will "survive" the scene change, so we can hope it's enough (10 is an arbitrary number, it's a tradeoff between "not too much" and "enough to avoid slowdown on most games").